src/main/java/org/semver4j/Semver.java

```diff
@@ -69,7 +80,8 @@ public static Semver parse(String version) {
      * @param version version to coerce
      * @return {@link Semver} if can coerce version, {@code null} otherwise
      */
-    public static Semver coerce(String version) {
+    @Nullable
+    public static Semver coerce(@Nullable final String version) {
```

Comment:
@DIG- @piotrooo Marking the input `version` as `@Nullable` is incorrect.
If you pass in `null` to this method, it throws a `NullPointerException` as it fails to create a regex matcher:
```
java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null
	at java.base/java.util.regex.Matcher.getTextLength(Matcher.java:1769)
	at java.base/java.util.regex.Matcher.reset(Matcher.java:415)
	at java.base/java.util.regex.Matcher.<init>(Matcher.java:252)
	at java.base/java.util.regex.Pattern.matcher(Pattern.java:1134)
	at org.semver4j.internal.Coerce.coerce(Coerce.java:20)
	at org.semver4j.Semver.coerce(Semver.java:89)
```

Reply:
Thanks for that good catch. I'll create a new issue with this.